api/server/acme/acme.go

```diff
@@ -0,0 +1,50 @@
+// Package acme abstracts away various ACME libraries
+package acme
+
+import (
+	"errors"
+	"net"
+
+	"golang.org/x/crypto/acme/autocert"
+)
+
+// ErrUnsupportedLibrary is returned by acme.New() if you
+// attempt to instantiate a library we don't support
+var ErrUnsupportedLibrary = errors.New("Unsupported Library")
+
+const (
+	// LibAutoCert is the acme library from golang.org/x/crypto/acme/autocert
+	LibAutoCert = "autocert"
+	// LibCertMagic is the acme library from github.com/mholt/certmagic
+	LibCertMagic = "certmagic"
+)
+
+// Library is an provider interface
+type Library interface {
+	NewListener([]string) (net.Listener, error)
+}
+
+// AutoCert is the ACME provider from golang.org/x/crypto/acme/autocert
+type AutoCert struct{}
```

Comment:
I would put the implementations under a sub directory for each

acme/autocert
acme/certmagic

And provide `type func New() acme.Acme` in each. We'd then accept the Acme interface as an option to a server and flag flip or switch between providers in the micro toolkit API and web packages.